### PR TITLE
Make kernel config smoketest more lenient

### PR DESCRIPTION
**Pull request by twz123**
## Description

It'll fail on systems which don't expose their kernel config at runtime, hence it may produce some unnecessary noise in those cases.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: internal/pkg/sysinfo/probes/linux/kernel_test.go [hunk 1]
```diff
@@ -118,7 +118,14 @@ func TestKConfigProber(t *testing.T) {
 
 		option, err := probeKConfig(ensureKConfig("I_CERTAINLY_DONT_EXIST"))
 		assert.Equal(t, option, kConfigUnknown)
-		assert.NoError(t, err)
+		switch err := err.(type) {
```
Comment (s0j) [inline internal/pkg/sysinfo/probes/linux/kernel_test.go:121]: Can you use `errors.Is()` here? Looks like you'd be able to remove a case with it.

Comment (s0j) [reply internal/pkg/sysinfo/probes/linux/kernel_test.go:121]: Oops, I meant `errors.As()`.  Something like this ..

```
var nce *noKConfigsFound
if errors.As(err, &nce) {
  t.Log("system doesn't ....")
} else {
  assert.NoError(t, err)
}
```

**Review by s0j — APPROVED** [review]
LGTM which ever way you choose, not blocking this.

File: internal/pkg/sysinfo/probes/linux/kernel_test.go (hunk 1 above)
Comment (twz123) [reply internal/pkg/sysinfo/probes/linux/kernel_test.go]: I've chosen a third option :-)

**Pull request merged**